MultiServer.py

```diff
@@ -1942,9 +1942,14 @@ async def process_client_cmd(ctx: Context, client: Client, args: dict):
             ctx.save()
             ctx.on_changed_hints(client.team, hint.finding_player)
             ctx.on_changed_hints(client.team, hint.receiving_player)
-        
+
         elif cmd == 'StatusUpdate':
-            update_client_status(ctx, client, args["status"])
+            if args["status"] == ClientStatus.CLIENT_GOAL and client.no_locations:
```

Comment:
Kind of just a feeling, but I think we'd get better shortcutting if this check was reversed since afaik the no location clients don't really update their status?

Reply:
I don't think I'm understanding the situation you're talking about, (also `/ready` exists)

Reply:
I'm saying
```suggestion
            if client.no_locations and args["status"] == ClientStatus.CLIENT_GOAL:
```
I did forget about /ready but I guess I have no idea how commonly people actually use that.

Reply:
sorry my not understanding was why that change would be better haha
isn't the assumption that most StatusUpdates would not be from no_location clients, thus adding an extra falsy check to every run?

Reply:
yeah and if it's false it skips the check on the other side of the and? I'm confused on the confusion here 😓

Reply:
lemme step back and itemize things because I for sure did get mixed around,
```
goal + no_loc		both true, unchanged order
goal + not no_loc	not False&True so faster
not goal + no_loc	now True&False so slower
not goal + not no_loc	both false, unchanged order
```
so the entire question depends on if there are more packets with a valid goal status update than packets with a not-goal update from no_loc clients
which is probably true as i'm hoping most slots end up goaling at least once, and /ready etc. aren't used multiple times per slot